contracts/governance/Staking/Staking.sol

```diff
@@ -298,21 +298,89 @@ contract Staking is
+     */
+    function cancelTeamVesting(
+        address vesting,
+        address receiver,
+        uint256 startFrom
+    ) external onlyAuthorized whenNotFrozen {
+        /// require the caller only for team vesting contract.
+        require(vestingRegistryLogic.isTeamVesting(vesting), "Only team vesting allowed");
+
+        _cancelTeamVesting(vesting, receiver, startFrom);
+    }
+
+    /**
+     * @notice Withdraws tokens from the staking contract and forwards them
+     * to an address specified by the token owner. Low level function.
+     * @dev Once here the caller permission is taken for granted.
+     * @param _vesting The vesting address.
+     * @param _receiver The receiving address.
+     * @param _startFrom The start value for the iterations.
+     * or just unlocked tokens (false).
      * */
-    function governanceWithdrawVesting(address vesting, address receiver)
-        public
-        onlyAuthorized
-        whenNotFrozen
-    {
-        vestingWhitelist[vesting] = true;
-        ITeamVesting(vesting).governanceWithdrawTokens(receiver);
-        vestingWhitelist[vesting] = false;
+    function _cancelTeamVesting(
+        address _vesting,
+        address _receiver,
+        uint256 _startFrom
+    ) private {
+        require(_receiver != address(0), "receiver address invalid");
+
+        ITeamVesting teamVesting = ITeamVesting(_vesting);
+
+        VestingConfig memory vestingConfig =
+            VestingConfig(
+                _vesting,
+                teamVesting.startDate(),
+                teamVesting.endDate(),
+                teamVesting.cliff(),
+                teamVesting.duration(),
+                teamVesting.tokenOwner()
+            );
+
+        uint96 tempStake;
+
+        /// @dev Usually we just need to iterate over the possible dates until now.
+        uint256 end;
+
+        uint256 defaultStart = vestingConfig.startDate + vestingConfig.cliff;
+
+        _startFrom = _startFrom >= defaultStart ? _startFrom : defaultStart;
 
-        emit VestingTokensWithdrawn(vesting, receiver);
+        /// @dev In the unlikely case that all tokens have been unlocked early,
+        ///   allow to withdraw all of them.
+        end = vestingConfig.endDate;
+
+        /// @dev max iterations need to be decreased by 1, otherwise the iteration will always be surplus by 1
+        uint256 totalIterationValue =
+            (_startFrom + (TWO_WEEKS * (getMaxVestingWithdrawIterations() - 1)));
+        uint256 adjustedEnd = end < totalIterationValue ? end : totalIterationValue;
+
+        /// @dev Withdraw for each unlocked position.
+        for (uint256 i = _startFrom; i <= adjustedEnd; i += TWO_WEEKS) {
+            /// @dev Read amount to withdraw.
+            tempStake = _getPriorUserStakeByDate(_vesting, i, block.number - 1);
+
+            if (tempStake > 0) {
+                /// @dev do governance direct withdraw for team vesting
+                _notSameBlockAsStakingCheckpoint(i);
```

Comment:
this function is checking checkpoints for `msg.sender`  
which is irrelevant here   
considering the function context can be removed

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/7f1c269ee40b1d2f3f864b07c574360839a9d270)